### PR TITLE
MINOR: Reuse KafkaConfig to create MetadataLogConfig

### PR DESCRIPTION
**Pull request by johnnychhsu**
## Context
These parameters are not in use, we should remove them

## Test
run ` ./gradlew clean core:test --tests KafkaConfigTest` and passed

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Review by brandboat — APPROVED** [review]
Good catch ! LGTM

**Comment by chia7712** [comment]
@johnnychhsu Instead of removing them, could you make `MetadataLogConfig` use those help methods? 

https://github.com/apache/kafka/blob/1b301b30207ed8fca9f0aea5cf940b0353a1abca/core/src/main/scala/kafka/MetadataLogConfig.scala#L38

**Comment by OmniaGM** [comment]
> @johnnychhsu Instead of removing them, could you make `MetadataLogConfig` use those help methods?
> 
> https://github.com/apache/kafka/blob/1b301b30207ed8fca9f0aea5cf940b0353a1abca/core/src/main/scala/kafka/MetadataLogConfig.scala#L38

This change also need to change `MetadataLogConfig` signature to use `KafkaConfig` instead of `AbstractConfig`. Which we actually do pass `KafkaConfig` in `KafkaRaftManager::buildMetadataLog`

**Review by OmniaGM — CHANGES_REQUESTED** [review]
The reason these aren't used is because `MetadataLogConfig` signature uses `AbstractConfig` and not `KafkaConfig`. We should update `MetadataLogConfig` class and make it use these methods as @chia7712 mentioned.

**Comment by johnnychhsu** [comment]
thanks for the review and suggestions @chia7712 @OmniaGM 
let me check and address it

**Comment by johnnychhsu** [comment]
> @johnnychhsu Instead of removing them, could you make `MetadataLogConfig` use those help methods?
> The reason these aren't used is because `MetadataLogConfig` signature uses `AbstractConfig` and not `KafkaConfig`. We should update `MetadataLogConfig` class and make it use these methods as @chia7712 mentioned.


@chia7712 @OmniaGM just updated according to the suggestions, thanks for the review!

**Review by OmniaGM — APPROVED** [review]
LGTM, thanks for updating this

**Comment by OmniaGM** [comment]
Just rebase to fix the conflicts

**Comment by johnnychhsu** [comment]
thanks for the prompt reply @OmniaGM !
just updated to resolve the conflict.

**Comment by chia7712** [comment]
@johnnychhsu Could you please fix conflicts?

**Comment by johnnychhsu** [comment]
thanks @chia7712!
just rebased and fixed the conflicts.

File: core/src/main/scala/kafka/MetadataLogConfig.scala [hunk 1]
```diff
@@ -32,13 +32,13 @@ final case class MetadataLogConfig(
...
-      config.getLong(KRaftConfigs.METADATA_MAX_RETENTION_BYTES_CONFIG),
-      config.getLong(KRaftConfigs.METADATA_MAX_RETENTION_MILLIS_CONFIG),
+      config.metadataLogSegmentBytes,
+      config.metadataLogSegmentMinBytes,
+      config.metadataLogSegmentMillis,
+      config.metadataRetentionBytes,
+      config.metadataRetentionMillis,
       maxBatchSizeInBytes,
       maxFetchSizeInBytes,
       ServerLogConfigs.LOG_DELETE_DELAY_MS_DEFAULT,
```
Comment (chia7712) [inline core/src/main/scala/kafka/MetadataLogConfig.scala:44]: Could you change line#45 to use `config.nodeId`?

**Review by chia7712 — COMMENTED** [review]
@johnnychhsu thanks for updated PR. it is almost there. one last comment left.

File: core/src/main/scala/kafka/MetadataLogConfig.scala (hunk 1 above)
Comment (johnnychhsu) [reply core/src/main/scala/kafka/MetadataLogConfig.scala:44]: thanks for the comment, nice catch!
let me update that

**Comment by johnnychhsu** [comment]
@chia7712
just updated, thanks for the review!

**Comment by chia7712** [comment]
```
./gradlew cleanTest :streams:test --tests SlidingWindowedKStreamIntegrationTest.shouldRestoreAfterJoinRestart :tools:test --tests MetadataQuorumCommandTest.testDescribeQuorumReplicationSuccessful --tests MetadataQuorumCommandTest.testDescribeQuorumStatusSuccessful --tests ListConsumerGroupTest.testListGroupCommandConsumerProtocol :storage:test --tests TopicBasedRemoteLogMetadataManagerTest.testNewPartitionUpdates --tests TransactionsWithTieredStoreTest.testSendOffsetsToTransactionTimeout :metadata:test --tests QuorumControllerTest.testBootstrapZkMigrationRecord --tests QuorumControllerTest.testNoOpRecordWriteAfterTimeout :trogdor:test --tests CoordinatorTest.testTaskRequestWithOldStartMsGetsUpdated :connect:mirror:test --tests IdentityReplicationIntegrationTest.testSyncTopicConfigs --tests MirrorConnectorsIntegrationBaseTest.testSyncTopicConfigs --tests MirrorConnectorsIntegrationTransactionsTest.testReplicateSourceDefault --tests MirrorConnectorsIntegrationExactlyOnceTest.testSyncTopicConfigs :core:test --tests ConsumerBounceTest.testClose --tests ConsumerBounceTest.testConsumptionWithBrokerFailures --tests PlaintextConsumerTest.testSimpleConsumption --tests PlaintextAdminIntegrationTest.testReplicaCanFetchFromLogStartOffsetAfterDeleteRecords --tests PlaintextAdminIntegrationTest.testDescribeConfigsForLog4jLogLevels --tests PlaintextConsumerTest.testPartitionsForAutoCreate --tests SaslMultiMechanismConsumerTest.testCoordinatorFailover --tests ControllerIntegrationTest.testTopicIdPersistsThroughControllerRestart --tests ZooKeeperClientTest.testBlockOnRequestCompletionFromStateChangeHandler --tests ReplicationQuotasTest.shouldBootstrapTwoBrokersWithFollowerThrottle
```
I don't observe related error and they pass on my local. will merge it

**Review by chia7712 — APPROVED** [review]

**Pull request merged**